test/test-execute/exec-restrictnamespaces-merge-and.service

```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=Test merging RestrictNamespaces= with AND
+
+[Service]
+RestrictNamespaces=mnt pid
+RestrictNamespaces=~mnt usr
```

Comment:
i think it would be good to add a test case whith these two rules:

```
RestrictNamespaces=mnt pid cgroup net pid uts ipc user
RestrictNamespaces=net
```

Which should result in `mnt pid cgroup net pid uts ipc user`